### ISSUE TITLE
False positives for testing-library/no-debugging-utils

### ISSUE THREAD

**Issue by sparker2-godaddy** [testing-library/eslint-plugin-testing-library#506]
### Plugin version

5.0.0

### ESLint version

8.3.0

### Node.js version

14.17.5

### npm/yarn version

yarn 1.22.15

### Operating system

macOS Big Sur, version 11.6.1

### Bug description

I created my own logging utils, one of which has a debug method (on the logger object) that only console logs messages if a debug env is set to true.
```ts
debug: (...messages: unknown[]): void => {
    if (process.env["REACT_APP_DEBUG_MODE"] === "true") {
      console.log(...messages);
    }
  },
```

I have excluded the required files as indicated in the setup of this library to avoid detecting anything outside of test files.  
```json
"overrides": [
    {
      "files": ["**/__tests__/**/*.[jt]s?(x)", "**/?(*.)+(spec|test).[jt]s?(x)"],
      "extends": ["plugin:testing-library/react"]
    }
  ],
```

When running my `yarn lint`, it gets picked up on a few places despite the exclusion.
example: `logger.debug("Test string", myObject, myNumber, myArray, true, undefined, null);`

![Screenshot 2021-11-22 at 15 18 07](https://user-images.githubusercontent.com/88374113/142868600-386c22fd-a74f-4fe5-89fc-dd7ee89fb7c3.png)

Am I missing something here in the override maybe? Even so, it's picking up the word "debug" outside of the scope of a react-testing-library context

### Steps to reproduce

1. Copy paste the code above for a simple logging wrapper.
2. Use it somewhere
3. Run the linter

### ESLint configuration
```json
{
  "extends": [
    "eslint:recommended",
    "plugin:import/errors",
    "plugin:react/recommended",
    "plugin:jsx-a11y/recommended",
    "plugin:react-hooks/recommended",
    "plugin:jest/recommended",
    "plugin:testing-library/react",
    "plugin:@typescript-eslint/recommended",
    "plugin:@typescript-eslint/recommended-requiring-type-checking",
    "prettier"
  ],
  "rules": {
    "react/prop-types": 0,
    "react/react-in-jsx-scope": 0,
    "react/display-name": 0,
    "@typescript-eslint/no-unsafe-assignment": 0,
    "@typescript-eslint/no-unsafe-member-access": 0,
    "@typescript-eslint/no-unnecessary-type-assertion": 0
  },
  "plugins": ["react", "import", "jsx-a11y", "jest", "testing-library", "@typescript-eslint"],
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "project": "./tsconfig.json",
    "ecmaVersion": 2021,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "env": {
    "es6": true,
    "browser": true,
    "node": true
  },
  "overrides": [
    {
      "files": ["**/__tests__/**/*.[jt]s?(x)", "**/?(*.)+(spec|test).[jt]s?(x)"],
      "extends": ["plugin:testing-library/react"]
    }
  ],
  "settings": {
    "react": {
      "version": "detect"
    },
    "import/parsers": {
      "@typescript-eslint/parser": [".ts", ".tsx"]
    },
    "import/resolver": {
      "typescript": {
        "alwaysTryTypes": true
      }
    }
  }
}
```

### Rule(s) affected

no-debugging-utils

### Do you want to submit a pull request to fix this bug?

No

**Comment by Belco90** [comment]
Hey @sparker2-godaddy, thanks for sharing. Let me help you with this.

You are getting this false positive in two different files:

1. `src/utils/logging-utils/logging-utils.test.ts`
2. `src/utils/monitoringutils/index.ts`

I'm afraid that you'll keep getting the error in 1) since it's a test file, so even if you set ESLint to run `eslint-plugin-testing-library` only on test files, that's a test file, so it will be linted by such plugin. However, 2) should be ignored with that config, right? Pay attention to your ESLint config file, since you have enabled `plugin:testing-library/react` twice: one in the `extends` property, and another one in the `overrides` property. You need to remove it from the first one: otherwise, you are still running `eslint-plugin-testing-library` over your whole codebase.

So that would fix 2), what about 1) then? You have several options:

1. You just disable `testing-library/no-debugging-utils` in `src/utils/monitoringutils/index.ts` file. That would do the trick, but you would have to the same workaround for other testing files you are still interested in testing stuff related to your debugging utils.
2. You switch off the [`testing-library/utils-module` option](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/migration-guides/v4.md#testing-libraryutils-module) so only utils imported from `@testing-library/<package>` are reported. Be careful with this one, since that would opt-out reporting custom Testing Library utils re-exported from another file (you'd have to specify the paths of the re-exported utils so they are reported too).

Give it a try and let me know how it goes!

**Comment by sparker2-godaddy** [comment]
Incredibly detailed and swift response with useful guidance. Thank you so much for the assistance.

I misread how the override works, and have updated it according to your suggestions with great success.

In regards to the second issue, I've opted to go for disabling that rule in those files only, as the logger should only really be used in the actual codebase and not in tests, apart from the test file that is testing the actual logger.

Thanks so much!

**Comment by Belco90** [comment]
I'm glad it worked!
